### PR TITLE
fixing host agent sampling and misc

### PR DESCRIPTION
**Pull request by goelayu**
1) fix min/max usage per host
2) run multiple processes per node
3) create multiple grpc client connections for every k hosts (currently commented out)
4) fix sampling of info bundles

### CODE REVIEW

File: go/deploy/actions/hostsim.go [hunk 1]
```diff
@@ -106,65 +106,71 @@ func RunHostAgentSims(c *pb.DeploymentConfig, remoteTopdir string, showOut bool)
...
 				}
-				return nil
-			})
+
+				eg.Go(func() error {
+					cmd := TracingCommand(
+						LogWithPrefix("run-host-agent-sims: "),
+						"ssh", n.GetExternalAddr(),
+						fmt.Sprintf("cat > %[1]s/configs/host-agent-sim-%[5]d.textproto && mkdir -p %[1]s/logs && ulimit -Sn unlimited && "+
+							"%[1]s/aux/host-agent-sim -m %[1]s/configs/host-agent-sim-%[5]d.mutex.out -c %[1]s/configs/host-agent-sim-%[5]d.textproto -o %[1]s/logs/host-agent-sim-%[5]d.csv -cluster-agent-addr %[2]s -dur %[4]s 2>&1 | tee %[1]s/logs/host-agent-sim-%[5]d.log; exit ${PIPESTATUS[0]}",
```
Comment (uluyol) [inline go/deploy/actions/hostsim.go:159]: You dropped the -start-time argument here,

File: go/deploy/actions/hostsim.go [hunk 2]
```diff
@@ -106,65 +106,71 @@ func RunHostAgentSims(c *pb.DeploymentConfig, remoteTopdir string, showOut bool)
...
 		}
 
 		const startHostID = 100001
 		numHosts := int32(0)
-		hostsPerNode := (int(c.C.GetNumHostsPerFg()) + len(c.Nodes) - 1) / len(c.Nodes)
+		nProcesses := 10
+		
 		for nidx, n := range c.Nodes {
 			n := n
+			hostsPerProcess := hostsPerNode/nProcesses
```
Comment (uluyol) [inline go/deploy/actions/hostsim.go:128]: Not sure this works if hostsPerNode doesn't divide evenly into nProcesses, but also I don't care too much. Maybe just add a check that there's 0 remainder and issue an error if so?

Compare with hostsPerNode that rounds up.

File: go/cmd/host-agent-sim/main.go [hunk 3]
```diff
@@ -33,6 +36,8 @@ func main() {
 
 	flag.Parse()
 
+	runtime.SetMutexProfileFraction(1)
```
Comment (uluyol) [inline go/cmd/host-agent-sim/main.go:39]: Can you make this conditional on *mutexProfPath != ""?

File: go/cmd/host-agent-sim/main.go [hunk 4]
```diff
@@ -92,6 +112,18 @@ func main() {
 	endTime := time.Now()
 	log.Printf("end-time: %s end-time-unix-ms: %d", endTime.In(time.UTC).Format(time.RFC3339Nano), unixMillis(endTime.In(time.UTC)))
 
+	m, err := os.Create(*mutexProfPath)
```
Comment (uluyol) [inline go/cmd/host-agent-sim/main.go:115]: Same as above.

**Comment by uluyol** [comment]
Commented on some minor issues but nothing worrying.

**Comment by goelayu** [comment]
@uluyol resolved the comments. 

Maybe one thing to do moving forward could be to add #processes per node as a value in the config file.

**Comment by uluyol** [comment]
Make sense if you find multi-process per node useful.

**Pull request merged**